packages/runtime/container-runtime/src/opLifecycle/outbox.ts

```diff
@@ -259,13 +259,40 @@ export class Outbox {
 			true /* disableGroupedBatching */,
 			resubmittingBatchId,
 		);
-		this.flushInternal(
+		this.flushInternalMainBatch(
 			this.mainBatch,
 			false /* disableGroupedBatching */,
 			resubmittingBatchId,
 		);
 	}
 
+	/**
+	 * Flushes the main batch to the ordering service. If the batch is empty, it will create an empty grouped batch
+	 * if BatchId is present (resubmit flow) and the runtime should send.
+	 */
+	private flushInternalMainBatch(
+		mainBatch: BatchManager,
+		disableGroupedBatching: boolean = false,
+		resubmittingBatchId?: BatchId,
+	) {
+		let clientSequenceNumber: number | undefined;
+		if (mainBatch.empty) {
+			if (resubmittingBatchId && this.params.shouldSend()) {
+				const emptyGroupedBatch = this.params.groupingManager.createEmptyGroupedBatch(
+					this.params.getCurrentSequenceNumbers().referenceSequenceNumber ?? 0,
+				);
+				clientSequenceNumber = this.sendBatch(emptyGroupedBatch);
+				this.params.pendingStateManager.onFlushBatch(
```

Comment:
You want to tell PSM about it even if `shouldSend` is false, just like in `flushInternal`. Otherwise it'll be dropped on the floor - but passing it back to PSM will ensure it gets resubmitted next time we connect.